pgxn/neon/walproposer.c

```diff
@@ -725,6 +725,7 @@ RecvAcceptorGreeting(Safekeeper *sk)
 			   sk->host, sk->port,
 			   sk->greetResponse.term, wp->propTerm);
 	}
+	pg_atomic_write_u64(&wp->api.get_shmem_state(wp)->mineLastElectedTerm, wp->propTerm);
```

Comment:
Wrong place, it must be set only after election (HandleElectedProposer or DetermineEpochStartLsn).

Reply:
Removed it. It gets set in DetermineEpochStartLsn already as well

Reply:
I had to readd this, otherwise tests fail with "could not acquire term 0"

Reply:
Hm, which test? I uncommented it and test_lr_with_slow_safekeeper still seems fine. For walproposer, SendProposerElected is called later than mineLastElectedTerm is set. Walsender indeed might try to fetch WAL before walproposer is elected, but then it should fail earlier with `no donor available`.

Reply:
test_lagging_sk failed for me

Reply:
Thanks!